### PR TITLE
docs: support code block titles

### LINKED ISSUE #21461: [internal] Allow code snippets on pw.dev with a custom title

### PR DESCRIPTION
**Pull request by mxschmitt**
Looks like this:

image

fixes https://github.com/microsoft/playwright/issues/21461

### CODE REVIEW

File: utils/markdown.js [hunk 1]
```diff
@@ -162,12 +163,20 @@ function buildTree(lines) {
 
     // Remaining items respect indent-based nesting.
     const [, indent, content] = /** @type {string[]} */ (line.match('^([ ]*)(.*)'));
+    const [codeLang, title] = (() => {
+      const lang = content.substring(3);
+      const match = lang.match(/ title="(.+)"/);
```
Comment (dgozman) [inline utils/markdown.js:168]: I think we should support any ordering between `title` and everything else, so perhaps split by space and find the title?

File: utils/markdown.js [hunk 2]
```diff
@@ -162,12 +163,20 @@ function buildTree(lines) {
 
     // Remaining items respect indent-based nesting.
     const [, indent, content] = /** @type {string[]} */ (line.match('^([ ]*)(.*)'));
+    const [codeLang, title] = (() => {
```
Comment (dgozman) [inline utils/markdown.js:166]: - Let's only do this inside if-block at line 173.
- Please extract a named function.

File: utils/markdown.js (hunk 1 above)
Comment (mxschmitt) [reply utils/markdown.js]: discussed offline, leaving it out for now.

File: utils/markdown.js [hunk 3]
```diff
@@ -162,12 +163,14 @@ function buildTree(lines) {
 
     // Remaining items respect indent-based nesting.
     const [, indent, content] = /** @type {string[]} */ (line.match('^([ ]*)(.*)'));
+    const [codeLang, title] = parseCodeBlockMetadata(content);
```
Comment (yury-s) [inline utils/markdown.js:166]: move this under `if` below?

**Review by yury-s — APPROVED** [review]

**Review by dgozman — APPROVED** [review]

**Pull request merged**